specs/refactor_gluster_integration_get_state_dump.adoc

```diff
@@ -0,0 +1,235 @@
+= Refactor Gluster Integration's state dump module
+
+
+The gluster get-state output is consumed by tendrl-gluster-integration. We need
+to re-factor the way Tendrl consumes and processes the get-state output.
+
+Notes:
+
+* In light of new changes to gluster get-state
+  http://review.gluster.org/#/c/15889
+
+
+== Problem description
+
+Current Tendrl way of consuming and processing the gluster get-state output
+considers only the volumes and bricks and pushes the details to central store.
+It needs to be modified to update the newly available volume options as well to
+central store.
+
+== Use Cases
+
+This addresses the use case of obtaining refreshed gluster cluster state and
+updating the same to central store.
+
+== Proposed change
+
+* Re-factor the code to handle the current output structure of get-state command
+related to volume options
+
+The current output from gluster get-state command is proposed to be of the
+format
+
+```
+[Global]
+MYUUID: 3780fd01-8047-402c-81df-e361a1935bb8
+op-version: 30900
+
+[Global options]
+
+[Peers]
+Peer1.primary_hostname: <fqdn1>
+Peer1.uuid: 3c94c637-cb3b-4580-b718-a7fc5abef4ef
+Peer1.state: Peer in Cluster
+Peer1.connected: Connected
+Peer1.othernames:
+Peer2.primary_hostname: <fqdn2>
+Peer2.uuid: 3f1459e9-9f68-4c6c-8a30-53c09188cd5f
+Peer2.state: Peer in Cluster
+Peer2.connected: Connected
+Peer2.othernames:
+
+[Volumes]
+Volume1.name: vol1
+Volume1.id: 5e24f6ed-239e-4961-ac47-1e9ed44aa7b1
+Volume1.type: Distribute
+Volume1.transport_type: tcp
+Volume1.status: Started
+Volume1.brickcount: 3
```

Comment:
You need to define all these options/attributes via Tendrl definitions because tendrl-api, tendrl-dashboard, tendrl-gluster-integration can discover these. 

If not defined in Tendrl definitions, we must not use them in our code. This is clear violation of our definitions driven approach

@brainfunked @nthomas-redhat @anivargi ^